### PR TITLE
feat: ContactQueryOptions and ContactBatch (Android)

### PR DESCRIPTION
**Pull request by pkar70**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?
ContactQueryOptions and ContactBatch are not implemented.

## What is the new behavior?
ContactQueryOptions and ContactBatch are implemented (for Android).

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [X] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information
Part of #2544, and continuation of #4081.
Also part of my app, CallsStat, published on Amazon.

Internal Issue (If applicable):

### CODE REVIEW

**Comment by pkar70** [comment]
Why it reports missing method? I include it in packageDiffIgnore...

File: src/Uno.UWP/ApplicationModel/Contacts/ContactQueryOptions.Android.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using Windows.Foundation;
+
+namespace Windows.ApplicationModel.Contacts
+{
+	public partial class ContactQueryOptions
+	{
+		internal ContactQuerySearchFields _whereToSearch;
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Contacts/ContactQueryOptions.Android.cs:12]: Use properties:

```suggestion
		internal ContactQuerySearchFields SearchFields { get; set; }
```

File: src/Uno.UWP/ApplicationModel/Contacts/ContactQueryOptions.Android.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		internal string _whatToSearch;
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Contacts/ContactQueryOptions.Android.cs:13]:
```suggestion
		internal string Text { get; set; }
```

File: src/Uno.UWP/ApplicationModel/Contacts/ContactBatch.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace Windows.ApplicationModel.Contacts
+{
+
+	public partial class ContactBatch
+	{
+
+		public IReadOnlyList<Contact> Contacts { get; }
+
+#if __ANDROID__
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Contacts/ContactBatch.cs:15]: There is no need to put this Android specific.

File: src/Uno.UWP/ApplicationModel/Contacts/ContactQueryOptions.Android.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
+#if __ANDROID__
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Contacts/ContactQueryOptions.Android.cs:1]: There is no need to set this android specific, the behavior is cross-platform.

File: src/Uno.UWP/ApplicationModel/Contacts/ContactBatch.cs (hunk 3 above)
Comment (pkar70) [reply src/Uno.UWP/ApplicationModel/Contacts/ContactBatch.cs:15]: It is used in Android implementation, and I know nothing about other implementations.
But ok.

File: src/Uno.UWP/ApplicationModel/Contacts/ContactQueryOptions.Android.cs (hunk 4 above)
Comment (pkar70) [reply src/Uno.UWP/ApplicationModel/Contacts/ContactQueryOptions.Android.cs:1]: Ok.

**Review by jeromelaban — APPROVED** [review]

**Comment by jeromelaban** [comment]
Thanks for the PR. Please squash your commits.

**Comment by pkar70** [comment]
So, this is approved, all checks have passed. Can it be merged?

**Review by davidjohnoliver — APPROVED** [review]

**Pull request merged**